x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LocalPhysicalPlanOptimizerTests.java

```diff
@@ -171,7 +277,7 @@ public void testCountAllWithEval() {
         var plan = plan("""
               from test | eval s = salary | rename s as sr | eval hidden_s = sr | rename emp_no as e | where e < 10050
             | stats c = count(*)
-            """);
+            """, basicMappingAnalyzer);
```

Comment:
Refactoring tests should simplify them not make them more complicated. 
All these tests expect an analyzer and should not require a basic piece of infrastructure to be plugged in.
Potentially overload the plan method to accept a different mapping and create an analyzer from it.